pkg/ratelimit/runner.go

```diff
@@ -144,8 +145,13 @@ func (cr *ConcurrentRunner) Start() {
 	}()
 }
 
-func (cr *ConcurrentRunner) run(task *Task, token *TaskToken) {
+func (cr *ConcurrentRunner) run(ctx context.Context, task *Task, token *TaskToken) {
 	start := time.Now()
+	select {
+	case <-ctx.Done():
+		return
```

Comment:
I think the race problem's main reason is the task of accessing the object without protection. here still cannot avoid that, we should let the task be safely run concurrency in task logic.

Reply:
Yes, indeed, but with the context, we can exit the goroutine ASAP.

Reply:
actually, I think it should pass context to the task, and then the task should consider exit ASAP by context cancel. here only prevent starting the new task.  the initial version has this design :)